src/util/permissions.tsx

```diff
@@ -0,0 +1,311 @@
+import { LxdIdentity, LxdPermission } from "types/permissions";
+import { OptionHTMLAttributes } from "react";
+import { LxdImage } from "types/image";
+
+export const defaultOption = {
+  disabled: true,
+  label: "Select an option",
+  value: "",
+};
+
+export const noneAvailableOption = {
+  disabled: true,
+  label: "None available",
+  value: "",
+};
+
+// the resource types comes from the openFGA authorisation model in lxd
+// ref: https://discourse.ubuntu.com/t/identity-and-access-management-for-lxd/41516
+export const resourceTypeOptions = [
+  {
+    ...defaultOption,
+  },
+  {
+    value: "server",
+    label: "Server",
+  },
+  {
+    value: "identity",
+    label: "Identity",
+  },
+  {
+    value: "group",
+    label: "Group",
+  },
+  {
+    value: "certificate",
+    label: "Certificate",
+  },
+  {
+    value: "project",
+    label: "Project",
+  },
+  {
+    value: "profile",
+    label: "Profile",
+  },
+  {
+    value: "instance",
+    label: "Instance",
+  },
+  {
+    value: "image",
+    label: "Image",
+  },
+  {
+    value: "image_alias",
+    label: "Image alias",
+  },
+  {
+    value: "storage_pool",
+    label: "Storage pool",
+  },
+  {
+    value: "storage_volume",
+    label: "Storage volume",
+  },
+  {
+    value: "storage_bucket",
+    label: "Storage bucket",
+  },
+  {
+    value: "network",
+    label: "Network",
+  },
+  {
+    value: "network_acl",
+    label: "Network ACL",
+  },
+  {
+    value: "network_zone",
+    label: "Network zone",
+  },
+];
+
+const sortOptions = (
+  a: OptionHTMLAttributes<HTMLOptionElement>,
+  b: OptionHTMLAttributes<HTMLOptionElement>,
+) => {
+  if (b.label === "Select an option") {
+    return 1;
+  }
+
+  return (a.label ?? "").localeCompare(b.label as string);
+};
+
+export const generateResourceAndEntitlementOptions = (
```

Comment:
I think this should be two functions, one to get resource options from the permissions and another one to get the entitlementOptions from the permissions.

Reply:
Yeah sure, I was hoping I could get all the options from one pass of permissions since the input here can get pretty big (e.g. each instance resource have 12 entitlements, so if you have 100 instances then that's 1200 permissions to process for a single selector). Looking at the code now I don't think it's possible especially since we have to sort the entitlements so splitting the function into two at least makes the code cleaner.